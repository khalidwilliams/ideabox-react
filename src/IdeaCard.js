import React from 'react';

const IdeaCard = ({id, title, description, deleteCard, isFavorite, toggleFavorite}) => {
  let favoriteClass;
  isFavorite ? favoriteClass = 'favorited' : favoriteClass = '';

  return (
    <section id={id} className={favoriteClass}>
      <h3>{title}</h3>
      <p>{description}</p>
      <div>
        <button className="favorite-btn" onClick={() => toggleFavorite(id)}>Favorite</button>
        <button className="delete-btn" onClick={() => deleteCard(id)}>Delete</button>
      </div>
    </section>
  );

};

export default IdeaCard;
