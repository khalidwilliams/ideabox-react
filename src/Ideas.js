import React from 'react';
import IdeaCard from './IdeaCard.js';

const Ideas = ({ideas, deleteCard, toggleFavorite}) => {
  let ideaCards = ideas.map(idea => {
    return <IdeaCard 
      key={idea.id}
      id={idea.id}
      title={idea.title}
      description={idea.description}
      isFavorite={idea.isFavorite}
      deleteCard={deleteCard}
      toggleFavorite={toggleFavorite}
    />
  });
  return (
    <main>{ideas}</main>
  );
};

export default Ideas;
