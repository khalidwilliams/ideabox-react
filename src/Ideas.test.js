import React from 'react';
import { shallow } from 'enzyme';
import Ideas from './Ideas.js';
import IdeaCard from './IdeaCard.js'

describe('Ideas Component', () => {
  describe('Rendering tests', () => {
    
    it('should be able to render with no ideas', () => {
      const deleteCardMock = jest.fn();
      let ideas = [
        {
          title:"Thought",
          id:1,
          description:"A Thought",
          deleteCard:deleteCardMock
        },
        {
          title:"Idea",
          id:2,
          description:"An Idea...",
          deleteCard:deleteCardMock
        }
      ];

      let shallowIdeas = shallow(<Ideas 
        ideas={ideas}  
      />); 
     expect(shallowIdeas).toBeDefined(); 
    });

    it('should be able to render with ideas', () => {
      const deleteCardMock = jest.fn();
      let ideas = [
        {
          title:"Thought",
          id:1,
          description:"A Thought",
          deleteCard:deleteCardMock
        },
        {
          title:"Idea",
          id:2,
          description:"An Idea...",
          deleteCard:deleteCardMock
        }
      ];

      let shallowIdeas = shallow(<Ideas 
        ideas={ideas}  
      />); 

      expect(shallowIdeas).toBeDefined();
    });
    it('will match the old snapshot', () =>{
      const deleteCardMock = jest.fn();
      let ideas = [
        {
          title:"Thought",
          id:1,
          description:"A Thought",
          deleteCard:deleteCardMock
        },
        {
          title:"Idea",
          id:2,
          description:"An Idea...",
          deleteCard:deleteCardMock
        }
      ];

      let shallowIdeas = shallow(<Ideas 
        ideas={ideas}  
      />); 
      expect(shallowIdeas).toMatchSnapshot();
    });
  });
  
});
