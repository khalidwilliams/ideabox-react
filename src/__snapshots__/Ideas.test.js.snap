// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ideas Component Rendering tests will match the old snapshot 1`] = `
<main>
  <Component />
  <Component />
</main>
`;
